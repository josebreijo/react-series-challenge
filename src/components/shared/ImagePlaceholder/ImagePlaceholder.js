import React from "react";

import { ColorBoxStyled } from "./ImagePlaceholder.styled";

const COLORS = [
  "#ff6b6b",
  "#48dbfb",
  "#1dd1a1",
  "#c8d6e5",
  "#5f27cd",
  "#222f3e",
  "#6ab04c"
];

const getRandomColor = index => COLORS[index || Math.floor(Math.random() * 7)];

const ImagePlaceholder = ({ height, width, index }) => (
  <ColorBoxStyled color={getRandomColor(index)} height={height} width={width} />
);

export default ImagePlaceholder;
