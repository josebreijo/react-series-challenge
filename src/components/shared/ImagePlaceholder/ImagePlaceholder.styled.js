import styled from "styled-components";

const ColorBoxStyled = styled.div`
  background-color: ${props => props.color};
  height: ${props => props.height}px;
  width: 25vw;
`;

export { ColorBoxStyled };
