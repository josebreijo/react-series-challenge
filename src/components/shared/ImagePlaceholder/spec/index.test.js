import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import ImagePlaceholder from "../";

describe("<ImagePlaceholder />", () => {
  it("renders properly", () => {
    const tree = renderer.create(<ImagePlaceholder index={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
