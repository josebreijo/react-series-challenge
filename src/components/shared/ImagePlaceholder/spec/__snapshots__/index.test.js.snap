// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImagePlaceholder /> renders properly 1`] = `
.c0 {
  background-color: #48dbfb;
  height: px;
  width: 25vw;
}

<div
  className="c0"
  color="#48dbfb"
  height={undefined}
  width={undefined}
/>
`;
